### DISCUSSION TITLE
What does word_timestamps do exactly? Can it produce word-segmented output?

### DISCUSSION THREAD

**Discussion by 7k50** [openai/whisper#1326] [Q&A] [answered]
Apologies for a basic question: what does  `word_timestamps="True"` do, exactly? 

I was hoping that `word_timestamps="True"` enables word-segmented output, like e.g. [WhisperX](https://github.com/m-bain/whisperX) does (example of its `.srt` output below):

```
1
00:00:09,564 --> 00:00:09,925
[SPEAKER_01]: Stop

2
00:00:10,245 --> 00:00:10,605
[SPEAKER_01]: that.
```

However, when running the below command with Whisper…

```whisper "{audio_path}" --model "{model}" --language "{language}" --output_dir "{output_dir_path}" --word_timestamps "True" --output_format "srt"```

…the console output is sentence-segmented:

```
[00:00.000 --> 00:03.480]  If you want to live longer and reduce your risk of different diseases,
[00:03.860 --> 00:08.340]  then it's important to maintain a physically active lifestyle and eat a good diet and get enough sleep.
```

…and the resulting `.srt` file is also sentence-segmented:

```
1
00:00:00,000 --> 00:00:03,480
If you want to live longer and reduce your risk of different diseases,

2
00:00:03,860 --> 00:00:08,340
then it's important to maintain a physically active lifestyle and eat a good diet and get enough sleep.
```

Presuming Whisper is working as intended and I am not configuring the command incorrectly, are we to understand that `word_timestamps="True"` just improves the accuracy of the timestamps on basis of words, but does not produce an output that is segmented on word-basis?

**Comment by ryanheise** [comment, marked as answer]
The console output is there to indicate transcription progress and is unaffected by this option.

The `srt` format is also unaffected by this because SRT is intended as a format for displaying subtitles, represented by whole lines of text (of an appropriate size that makes them easily readable when displayed below the media).

the `--word_timestamps` option asks Whisper to internally infer timestamps for each word, but if you want to see them outputted, then you also need to select an output format suitable for words. For example, `--output_format json` will include a `"words"` key in each segment containing a list of the individual words and their timestamps for that segment.

Also, if you truly want SRT (i.e. subtitles), there is a variation on this known as Karaoke subtitles, which is possible when enabling both the `--output_format srt` option AND the `--highlight_words True` option. You will then get an SRT file with a separate entry for each word timestamp, but each entry will also contain the whole line surrounding it (so that it still respects the "subtitle" intention of SRT) with the current word highlighted with `<u>word</u>`.
